app/filtering.js

```diff
@@ -57,6 +57,51 @@ const registeredSessions = {}
  */
 const permissionCallbacks = {}
 
+/**
+ * Cache of URLs mapped their site settings.
+ * This gets rebuilt when site settings or a bravery setting is changed.
+ */
+let braverySettingsForUrl = {}
+
+const getBraverySettingsForUrl = (url, appState, isPrivate) => {
+  if (braverySettingsForUrl[url]) {
+    return braverySettingsForUrl[url]
+  }
+  const savedSettings = siteSettings.getSiteSettingsForURL(appState.get('siteSettings'), url)
+  const tempSettings = siteSettings.getSiteSettingsForURL(appState.get('temporarySiteSettings'), url)
+
+  let braverySettings = siteSettings.activeSettings(savedSettings, appState, appConfig)
+  if (isPrivate && tempSettings) {
+    braverySettings = siteSettings.activeSettings(tempSettings, appState, appConfig)
+  }
+  braverySettingsForUrl[url] = braverySettings
+  return braverySettings
+}
+
+const settingsKeys = ['siteSettings', 'temporarySiteSettings', 'settings']
+const appStoreChangeCallback = function (diffs) {
```

Comment:
I think instead of an app store listener on all events here we should handle the events that change site settings in a reducer and call into a cache module (braverySettingsCache) managed from the reducer and used in filtering.